mm2src/coins/lightning/ln_utils.rs

```diff
@@ -443,3 +824,201 @@ async fn ln_node_announcement_loop(
+                }
+            }
+        },
+        None => {
+            return MmError::err(ConnectToNodeError::ConnectionError(format!(
+                "Failed to connect to node: {}",
+                pubkey
+            )))
+        },
+    }
+
+    Ok(ConnectToNodeRes::ConnectedSuccessfully(
+        pubkey.to_string(),
+        node_addr.to_string(),
+    ))
+}
+
+async fn connect_to_node_loop(ctx: MmArc, pubkey: PublicKey, node_addr: SocketAddr, peer_manager: Arc<PeerManager>) {
+    for node_pubkey in peer_manager.get_peer_node_ids() {
+        if node_pubkey == pubkey {
+            log::info!("Already connected to node: {}", node_pubkey);
+            return;
+        }
+    }
+
+    loop {
+        if ctx.is_stopping() {
+            break;
+        }
+
+        match connect_to_node(pubkey, node_addr, peer_manager.clone()).await {
+            Ok(res) => {
+                log::info!("{}", res.to_string());
+                break;
+            },
+            Err(e) => log::error!("{}", e.to_string()),
+        }
+
+        Timer::sleep(TRY_RECONNECTING_TO_NODE_INTERVAL as f64).await;
+    }
+}
+
+pub fn open_ln_channel(
+    node_pubkey: PublicKey,
+    amount_in_sat: u64,
+    events_id: u64,
+    announce_channel: bool,
+    channel_manager: Arc<ChannelManager>,
+) -> OpenChannelResult<[u8; 32]> {
+    // TODO: get user_config from context when it's added to it
+    let mut user_config = UserConfig::default();
+    user_config
+        .peer_channel_config_limits
+        .force_announced_channel_preference = false;
+    user_config.channel_options.announced_channel = announce_channel;
+
+    // TODO: push_msat parameter
+    channel_manager
+        .create_channel(node_pubkey, amount_in_sat, 0, events_id, Some(user_config))
+        .map_to_mm(|e| OpenChannelError::FailureToOpenChannel(node_pubkey.to_string(), format!("{:?}", e)))
+}
+
+// Generates the raw funding transaction with one output equal to the channel value.
+async fn generate_funding_transaction(
+    ctx: MmArc,
+    channel_value_satoshis: u64,
+    output_script: Script,
+    user_channel_id: u64,
+    coin: Arc<UtxoStandardCoin>,
+) -> Result<Transaction, WithdrawError> {
+    let utxo_coin = coin.as_ref().as_ref();
+    let addr = SegwitAddress::new(
+        &AddressHashEnum::WitnessScriptHash(output_script[2..].into()),
+        utxo_coin
+            .conf
+            .bech32_hrp
+            .clone()
+            .expect("Coin should have a bech32_hrp"),
+    )
+    .to_string();
+    let amount = big_decimal_from_sat(channel_value_satoshis as i64, utxo_coin.decimals);
+
+    let lightning_ctx = LightningContext::from_ctx(&ctx).unwrap();
+    let (fee, max) = {
+        let mut funding_tx_params = lightning_ctx.funding_tx_params.lock().await;
+        funding_tx_params.remove(&user_channel_id).unwrap_or((None, false))
+    };
+
+    let withdraw_req = WithdrawRequest::new(utxo_coin.conf.ticker.clone(), addr, amount, max, fee);
```

Comment:
This way changes in `withdraw` may also result in `generate_funding_transaction` change, which is undesirable IMHO. I would prefer to see direct usage of UtxoTxBuilder here. Is it feasible to make it generic over the returned transaction type?